### PR TITLE
chore: Add CODEOWNERS.

### PR DESCRIPTION
**Pull request by justin-determined-ai**
## Commentary (optional)

I did a first pass at ownership files/folders, but I'm likely off on some. Could folks please take a look and modify the file when necessary? Also, there are some features (like model registry) that are spread out over many files; I've tagged people who I might need input from on the review.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Scheduler
+master/internal/scheduler/    @aaron276h @rb-determined-ai
+
+# Provisioner
+master/internal/provisioner/  @shiyuann
+
+# Model registry
+# @eecsliu @katport
```
Comment (justin-determined-ai) [inline CODEOWNERS:8]: @eecsliu @katport is there a main set of files for this? I found this stuff distributed across many parts of the code

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+# Experiment/trial actors
+master/internal/trial.go      @rb-determined-ai @dzhu @aaron276h
+master/internal/experiment.go @rb-determined-ai @dzhu @aaron276h
+
+# Actor infrastructure
+master/pkg/actor              @dzhu
+
+# D-train
+# @aaron276h @armandmcqueen
```
Comment (justin-determined-ai) [inline CODEOWNERS:21]: @aaron276h @armandmcqueen is there a main set of files for this? I found this stuff distributed (heh) across many parts of the code

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+# HP Search
+master/pkg/searcher/          @stoksc @liamcli
+
+# Checkpoint export
+#
```
Comment (justin-determined-ai) [inline CODEOWNERS:27]: ditto above; also, there's no owner currently set.

File: CODEOWNERS (hunk 2 above)
Comment (aaron276h) [reply CODEOWNERS:21]: Its a bit all over the place, https://github.com/determined-ai/horovod, a good amount of the harness and some scheduler stuff.

Comment (justin-determined-ai) [reply CODEOWNERS:21]: do you have a set of candidate files/folders we should start with?

File: CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+# Commands/shell
+/master/internal/command/command_manager.go      @dzhu
+/master/internal/command/shell_manager.go        @dzhu
+
+# CI infrastructure
+/.circleci/                    @aaron276h @rb-determined-ai @dzhu
+
+# Build/packaging
+packaging/                     @dzhu @rb-determined-ai
```
Comment (justin-determined-ai) [inline CODEOWNERS:64]: note I omitted the leading slash on purpose to capture _any_ folder named `packaging`.

File: CODEOWNERS (hunk 2 above)
Comment (aaron276h) [reply CODEOWNERS:21]: https://github.com/horovod/horovod
https://github.com/determined-ai/determined/blob/master/harness/determined/horovod.py
https://github.com/determined-ai/determined/blob/master/harness/determined/ipc.py
https://github.com/determined-ai/determined/blob/master/harness/determined/layers/_worker_process.py
https://github.com/determined-ai/determined/blob/master/harness/determined/_trial_controller.py
https://github.com/determined-ai/determined/blob/master/harness/determined/exec/worker_process.py

File: CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+# Provisioner
+/master/internal/provisioner/  @shiyuann
+
+# Model registry
+/master/pkg/model              @eecsliu @katport
+/proto/pkg/modelv1             @eecsliu @katport
+
+# WebUI
+/webui/                        @hkang1 @hamidzr
```
Comment (hamidzr) [inline CODEOWNERS:12]: maybe we separate the sub-WebUI modules :thinking:

**Review by hamidzr — APPROVED** [review]
the API and WebUI portion LGTM.

File: CODEOWNERS (hunk 5 above)
Comment (justin-determined-ai) [reply CODEOWNERS:12]: Into what? And who would own which part?

Comment (hamidzr) [reply CODEOWNERS:12]: maybe a separate line for WebUI e2e tests? as-is also works.

**Pull request merged**